sklearn/ensemble/tests/test_forest.py

```diff
@@ -1456,13 +1457,24 @@ def check_warm_start_oob(name):
         bootstrap=True,
         oob_score=False,
     )
-    est_3.fit(X, y)
-    assert not hasattr(est_3, "oob_score_")
 
-    est_3.set_params(oob_score=True)
-    ignore_warnings(est_3.fit)(X, y)
+    # Patch _set_oob_score_and_attributes() to track OOB computation
```

Comment:
I think extending this test complicates the original test a little too much. I prefer adding another test:

```python
def test_oob_not_computed_twice(monkeypatch):
    """Check that oob_score is not computed twice when warm_start=True."""
    est = RandomForestClassifier(n_estimators=10, oob_score=True, warm_start=True)

    mock = Mock(side_effect=est._set_oob_score_and_attributes)
    monkeypatch.setattr(est, "_set_oob_score_and_attributes", mock)

    est.fit(X, y)

    with pytest.warns(UserWarning, match="Warm-start fitting without increasing"):
        est.fit(X, y)

    mock.assert_called_once()
```

where `from unittest.mock import Mock`.